src/main/java/io/cdap/plugin/gcp/common/GCPErrorDetailsProvider.java

```diff
@@ -93,6 +103,52 @@ private ProgramFailureException getProgramFailureException(HttpResponseException
       pair.getErrorType(), true, e);
   }
 
+
+  /**
+   * Get a ProgramFailureException with the given error
+   * information from {@link IllegalArgumentException}.
+   *
+   * @param e The IllegalArgumentException to get the error information from.
+   * @return A ProgramFailureException with the given error information.
+   */
+  private ProgramFailureException getProgramFailureException(IllegalArgumentException e, ErrorContext errorContext) {
+    String errorMessage = e.getMessage();
+    String subCategory = "IllegalArgument";
+    String errorMessageFormat = "Error occurred in the phase: '%s'. Error message: %s";
+    return ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategoryEnum.PLUGIN, subCategory), errorMessage,
+      String.format(errorMessageFormat, errorContext.getPhase(), errorMessage), ErrorType.USER, true, e);
+  }
+
+  /**
+   * Get a ProgramFailureException with the given error
+   * information from {@link IllegalStateException}.
+   *
+   * @param e The IllegalStateException to get the error information from.
+   * @return A ProgramFailureException with the given error information.
+   */
+  private ProgramFailureException getProgramFailureException(IllegalStateException e, ErrorContext errorContext) {
```

Comment:
Similarly, how is `IllegalStateException` a `user` error?

Reply:
An `IllegalStateException` usually represents a programming or configuration error in the code. Error Type `System` might be better fit as it suggests something went wrong with the program's state or lifecycle that's under the application's control rather than being an external client issue.

Reply:
In code there are places where we can reach `IllegalStateException` by incorrect user inputs, either we should change those.

Reply:
https://github.com/data-integrations/google-cloud/blob/601f62bd0005255e75bff19db8711a8898358d49/src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java#L192

Reply:
this above should be changed to IllegalArgumentException right ?

Reply:
Changed `IllegalStateException` to `SYSTEM`

Reply:
> this above should be changed to IllegalArgumentException right ?

right.

Reply:
Updated !